### PR TITLE
BZ2106748: Add a note regarding x509 certificate validation error

### PR DESCRIPTION
**Pull request by xenolinux**
Add a note regarding x509 certificate validatio error

Version(s): 4.8+

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2106748

Link to docs preview: https://55091--docspreview.netlify.app/openshift-enterprise/latest/operators/admin/olm-configuring-proxy-support.html#olm-inject-custom-ca_olm-configuring-proxy-support

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/olm-injecting-custom-ca.adoc [hunk 1]
```diff
@@ -68,3 +68,8 @@ spec:
 <4> `ca-bundle.crt` is required as the config map key.
 <5> `tls-ca-bundle.pem` is required as the config map path.
 <6> Create a `trusted-ca` volume mount.
++
+[NOTE]
+====
+If an Operator pod fails with the x509 certificate validation error, set the  `mountPath` as `/etc/ssl/certs`.
```
Comment (jianzhangbjz) [inline modules/olm-injecting-custom-ca.adoc:74]: Not all x509 errors can be fixed by this operation.

Comment (xenolinux) [reply modules/olm-injecting-custom-ca.adoc:74]: @jianzhangbjz Can we add any additional information or can we rearrange the sentence to something like - 
`You can avoid some x509 certificate validation errors by setting the `mountPath` as `/etc/ssl/certs`. ?

**Comment by xenolinux** [comment]
@Xia-Zhao-rh Could you please review the new changes?

File: modules/olm-injecting-custom-ca.adoc [hunk 2]
```diff
@@ -68,3 +68,8 @@ spec:
 <4> `ca-bundle.crt` is required as the config map key.
 <5> `tls-ca-bundle.pem` is required as the config map path.
 <6> Create a `trusted-ca` volume mount.
++
+[NOTE]
+====
+Sometimes deployment of an Operator fails to validate the authority and throws the `x509 certificate signed by unknown authority` error, even after injecting a custom CA by using the subscription of an Operator. In this case, it is recommended to inject the entire CA chain. If the deployment of a Operator still fails then you can set the `mountPath` as `/etc/ssl/certs` for trusted-ca by using the subscription of an Operator.
```
Comment (kelbrown20) [inline modules/olm-injecting-custom-ca.adoc:74]:
```suggestion
Deployments of an Operator can fail to validate the authority and display a `x509 certificate signed by unknown authority` error. This error can occur even after injecting a custom CA when using the subscription of an Operator. In this case, it is recommended to inject the entire CA chain. If the deployment of a Operator still fails, then you can set the `mountPath` as `/etc/ssl/certs` for trusted-ca by using the subscription of an Operator.
```

Comment (kelbrown20) [inline modules/olm-injecting-custom-ca.adoc:74]: `In this case, it is recommended to inject the entire CA chain.`
I'm not sure on the context, but it seems like this sentence needs a prepositional phrase at the end. Like `it is recommended to inject the entire CA chain to the Operator`, or something with the correct term.

**Review by kelbrown20 — COMMENTED** [review]
Hello! Just some minor suggestions, great work!

File: modules/olm-injecting-custom-ca.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/olm-injecting-custom-ca.adoc]: Addressed

Comment (xenolinux) [reply modules/olm-injecting-custom-ca.adoc]: I confirmed with the bug reporter, we can exclude the whole sentence. 

Addressed

Comment (xenolinux) [reply modules/olm-injecting-custom-ca.adoc]: Addressed

**Comment by michaelryanpeter** [comment]
/lgtm

@xenolinux I am just cherrypicking to 4.9+ because 4.8 is no longer recieving bug fixes.

**Pull request merged**